### DISCUSSION TITLE
How to config db-scheduler to be able work with AWS architecture.

### DISCUSSION THREAD

**Discussion by CamilYed** [kagkarlsson/db-scheduler#401] [Q&A]
Hello.
I have 3 regions in my AWS architecture:
Region A
Region B
Region C
In each of these regions, 3 instances of the application.

I would like one instance in each region to be able to perform a task.

What should my Sheduler configuration look like.

```java
@Bean
    SchedulerBuilder schedulerBuilder(
            @Qualifier("dbSchedulerDataSource") DataSource dbSchedulerDataSource,
            ExecutorService dbSchedulerExecutorService
    ) {
        return Scheduler.create(dbSchedulerDataSource)
                .commitWhenAutocommitDisabled(true)
                .schedulerName() // TODO <- what i should provide here
                .executorService(dbSchedulerExecutorService);
    }
```

in my application.properties i have two property:
region - name of the aws region
instance - name of the instance in region

Database is shared between regions.

**Comment by kagkarlsson** [comment]
As long as the database is shared between application instances, all instances in all az/regions will compete for the tasks
